### PR TITLE
feat(toast): add swipeGesture

### LINKED ISSUE #21769: feat: Swipe to dismiss Toasts
# Feature Request
Toast can optionally be dismissed via swiping the toast.

**Ionic version:**

[x] **4.x**
[x] **5.x**

**Describe the Feature Request**

Toasts should be dismissible via swiping. Many modern mobile apps that show messages similar to Toasts in Ionic are easily dismissible via swiping left or right on the toast.

**Describe Preferred Solution**

A swipeToClose boolean option that could be added to the opts parameter in the Toast Controller's .create() function. Set to false by default.

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: resolves #21769

---------

## What is the current behavior?

Toast does not support swipe gestures to dismiss.

## What is the new behavior?

- Added a `swipeGesture` property that allows users to swipe toasts closed.
Note: This is a combination of previous PRs https://github.com/ionic-team/ionic-framework/pull/28380 and https://github.com/ionic-team/ionic-framework/pull/28402

⚠️ There is a visual glitch on iOS where dragging and having the toast animate back to its opened position causes a flicker. This is an iOS 17 regression and is being tracked in https://github.com/ionic-team/ionic-framework/issues/28467. This bug has been reported to and confirmed by Apple.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

⚠️ Give co-author credit to author in https://github.com/ionic-team/ionic-framework/pull/23124

### CODE REVIEW

**Comment by thetaPC** [comment]
Dev build: 7.5.4-dev.11699469920.1af6ea12

**Review by thetaPC — APPROVED** [review]
LGTM

File: core/src/components/toast/animations/utils.ts [hunk 1]
```diff
@@ -88,3 +88,15 @@ function warnIfAnchorIsHidden(positionAnchor: HTMLElement, toast: HTMLElement) {
     );
   }
 }
+
+/**
+ * Returns the top offset required to place
+ * the toast in the middle of the screen.
+ * Only needed when position="toast".
+ * @param toastHeight: The height of the ion-toast element
```
Comment (sean-perkins) [inline core/src/components/toast/animations/utils.ts:96]:
```suggestion
 * @param toastHeight - The height of the ion-toast element
```

File: core/src/components/toast/animations/utils.ts [hunk 2]
```diff
@@ -88,3 +88,15 @@ function warnIfAnchorIsHidden(positionAnchor: HTMLElement, toast: HTMLElement) {
...
+ * @param wrapperHeight: The height of the .toast-wrapper element
```
Comment (sean-perkins) [inline core/src/components/toast/animations/utils.ts:97]:
```suggestion
 * @param wrapperHeight - The height of the .toast-wrapper element
```

**Review by sean-perkins — APPROVED** [review]
Looks good! Only 2 super minor JSDOC formatting suggestions.

**Pull request merged**